File: DSA-python/String/jewelsAndStones.py
# this is the solution for the brute force approach

class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        """
        :type jewels: str
        :type stones: str
        :rtype: int
        """
        count = 0
        for i in range(len(stones)):
            if stones[i] in jewels:
                count+=1
        return count

# another updated better solution would be

class Solution(object):
    def numJewelsInStones(self, jewels, stones):
        """
        :type jewels: str
        :type stones: str
        :rtype: int
        """
        count = 0
        jewel = set(jewels)
        for i in range(len(stones)):
            if stones[i] in jewel :
                count+=1
        return count

        

File: DSA-js/String/jewelsAndStones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function(jewels, stones) {
    let count = 0;
    let jewelss = new Set()
     for(let i=0;i<jewels.length;i++)
     {
        jewelss.add(jewels[i]);
     }
     for(let j=0;j<stones.length;j++)
     {
        if(jewelss.has(stones[j]))
        {
            count++;
        }
     }
     return count

};
// hello